src/mongo/tree/MongoDatabaseTreeItem.ts

```diff
@@ -128,20 +128,22 @@ export class MongoDatabaseTreeItem extends AzureParentTreeItem<IMongoTreeRoot> {
 		return new MongoCollectionTreeItem(this, newCollection);
 	}
 
-	executeCommandInShell(command: MongoCommand, context: IActionContext): Thenable<string> {
+	private executeCommandInShell(command: MongoCommand, context: IActionContext): Thenable<string> {
 		context.properties["executeInShell"] = "true";
-		return this.getShell().then(shell => shell.exec(command.text));
+		return this.getShell().then(shell => shell.executeScript(command.text));
```

Comment:
One of the biggest confusions I had about this code was that `createShell` implied it was keeping a shell around. In reality, it's only ever executing one script and then throwing it away. Can we remove `createShell` and move that code into this function?